astarte/device/device.py

```diff
@@ -143,6 +145,45 @@ def add_interface(self, interface_definition: dict) -> None:
         """
         self.__introspection.add_interface(interface_definition)
 
+    def add_interface_from_file(self, interface_file: Path):
+        """
+        Adds an interface to the device
+
+        This will add an interface definition to the device. It has to be called before
+        :py:func:`connect`, as it will be used for building the device introspection.
+
+        Parameters
+        ----------
+        interface_file : Path
+            An absolute path to an Astarte interface json file.
+        """
+        if not interface_file.is_file():
+            raise FileNotFoundError("Non-existent interface file.")
+        if interface_file.suffix != ".json":
+            raise TypeError("Incorrect format for the interface file.")
+        with open(interface_file, "r", encoding="utf-8") as interface_fp:
+            self.__introspection.add_interface(json.load(interface_fp))
+
+    def add_interfaces_from_dir(self, interfaces_dir: Path):
+        """
+        Adds a series of interfaces to the device
+
+        This will add all the interfaces contained in the provided folder to the device.
+        It has to be called before :py:func:`connect`, as it will be used for building the device
+        introspection.
+
+        Parameters
+        ----------
+        interfaces_dir : Path
+            An absolute path to an a folder containing some Astarte interface .json files.
+        """
+        if not interfaces_dir.is_dir():
+            raise FileNotFoundError("Non-existent interface directory.")
```

Comment:
```suggestion
        if not interfaces_dir.exists():
            raise FileNotFoundError(f'"{interfaces_dir}" does not exist')
        if not interfaces_dir.is_dir():
            raise NotADirectoryError(f'"{interfaces_dir}" is not a directory')
```
I suggest using "FileNotFoundError" if the path does not exist, a "NotADirectoryError" if the path is not a directory, or, even better, adding both checks.